libs/community/langchain_community/document_loaders/unstructured.py

```diff
@@ -183,60 +195,40 @@ def _get_elements(self) -> List:
     def _get_metadata(self) -> dict:
         return {"source": self.file_path}
 
+    def _post_process_elements(self, elements: list) -> list:
+        """Apply post processing functions to extracted unstructured elements.
 
-def get_elements_from_api(
-    file_path: Union[str, List[str], Path, List[Path], None] = None,
-    file: Union[IO, Sequence[IO], None] = None,
-    api_url: str = "https://api.unstructured.io/general/v0/general",
-    api_key: str = "",
-    **unstructured_kwargs: Any,
-) -> List:
-    """Retrieve a list of elements from the `Unstructured API`."""
-    if is_list := isinstance(file_path, list):
-        file_path = [str(path) for path in file_path]
-    if isinstance(file, collections.abc.Sequence) or is_list:
-        from unstructured.partition.api import partition_multiple_via_api
-
-        _doc_elements = partition_multiple_via_api(
-            filenames=file_path,
-            files=file,
-            api_key=api_key,
-            api_url=api_url,
-            **unstructured_kwargs,
-        )
-
-        elements = []
-        for _elements in _doc_elements:
-            elements.extend(_elements)
-
+        Post processing functions are str -> str callables passed
+        in using the post_processors kwarg when the loader is instantiated.
+        """
+        for element in elements:
+            for post_processor in self.post_processors:
+                element.apply(post_processor)
         return elements
-    else:
-        from unstructured.partition.api import partition_via_api
-
-        return partition_via_api(
-            filename=str(file_path) if file_path is not None else None,
-            file=file,
-            api_key=api_key,
-            api_url=api_url,
-            **unstructured_kwargs,
-        )
-
 
-class UnstructuredAPIFileLoader(UnstructuredFileLoader):
+@deprecated(
+    since="0.2.8",
+    removal="0.4.0",  # TODO (jennings): determine when to set removal
+    alternative_import="langchain_unstructured.UnstructuredSDKFileLoader",
+)
```

Comment:
@MthwRobinson, is version 0.4.0 a reasonable benchmark for removing these loaders?

Reply:
Yeah that seems like plenty of warning!